tests/functional/conftest.py

```diff
@@ -0,0 +1,58 @@
+import os
+
+import pytest
+import webtest
+
+from tests.simple_server import serve_content
+from viahtml.app import Application
+
+
+@pytest.fixture(scope="session")  # pragma: no cover
+def app(with_environ):
```

Comment:
```suggestion
@pytest.mark.usefixtures("environ")
def app():
```

Alternatively, I think the `environ` / `with_environ` fixture below could just be `autouse=True`? If `app` depends on it then I think it's getting used in every test. Declare the fact that `environ` is always used in `environ`'s own `@pytest.fixture(...)` instead of indirectly by making `app` depend on it

Reply:
The environ changes must be in place before the app is created. So we must force the order or it won't start.